src/core/qgsproject.h

```diff
@@ -1154,6 +1156,19 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     QgsMapLayer *takeMapLayer( QgsMapLayer *layer ) SIP_TRANSFERBACK;
 
+    /**
+     * Returns the main annotation layer associated with the project.
+     *
+     * This layer is always present in projects, and will always be rendered
+     * above any other map layers during map render jobs.
+     *
+     * It forms the default location to place new annotation items which
+     * should appear above all map layers.
+     *
+     * \since QGIS 3.16
+     */
+    QgsAnnotationLayer *mainAnnotationLayer();
```

Comment:
I think I'd just go for annotationLayer() and remove the main prefix.

Reply:
I'm not so keen on that -- I wanted a clear distinction between this special layer and any "normal" annotation layers which may be present in the project (via the standard QgsProject map layer storage methods). 

An alternative could be "defaultAnnotationLayer", but I'd still like something which conveys the concept that it's "special" and works different to normal layers (because it's always drawn on top)

Reply:
persistentAnnotationLayer , topAnnotationLayer ? I just feel that the main prefix doesn't add much context here.

In any case, I see your point and I'll get over it if you merge as is :wink: go with what you think is best.